ReplayKitExample/ReplayKitExample/ViewController.swift

```diff
@@ -307,71 +304,48 @@ class ViewController: UIViewController {
 
-        let videoCodec = Settings.shared.videoCodec ?? Vp8Codec()!
-        let (encodingParams, outputFormat) = ReplayKitVideoSource.getParametersForUseCase(codec: videoCodec,
-                                                                                          isScreencast: !ViewController.kDownscaleBuffers,
-                                                                                       telecineOptions:options)
-        videoSource?.requestOutputFormat(outputFormat)
-
-        recorder.startCapture(handler: { (sampleBuffer, type, error) in
-            if error != nil {
-                print("Capture error: ", error as Any)
-                return
-            }
-
-            switch type {
-            case RPSampleBufferType.video:
-                self.videoSource?.processFrame(sampleBuffer: sampleBuffer)
-                break
-            case RPSampleBufferType.audioApp:
-                break
-            case RPSampleBufferType.audioMic:
-                // We use `TVIDefaultAudioDevice` to capture and playback audio for conferencing.
-                break
-            }
-
-        }) { (error) in
+        appScreenSource?.startCapture { _, error in
             if error != nil {
                 print("Screen capture error: ", error as Any)
             } else {
                 print("Screen capture started.")
             }
-            DispatchQueue.main.async {
-                self.conferenceButton?.isEnabled = true
-                if error != nil {
-                    self.broadcastButton.isEnabled = true
-                    self.broadcastButton.setTitle(ViewController.kStartBroadcastButtonTitle, for: UIControl.State.normal)
-                    self.broadcastPickerView?.isHidden = false
-                    self.broadcastButton.isHidden = false
-                    self.conferenceButton?.setTitle(ViewController.kStartConferenceButtonTitle, for:.normal)
-                    self.infoLabel?.isHidden = false
-                    self.infoLabel?.text = error!.localizedDescription
-                    self.videoSource = nil
-                    self.screenTrack = nil
-                } else {
-                    self.conferenceButton?.setTitle(ViewController.kStopConferenceButtonTitle, for:.normal)
-                    self.spinner.startAnimating()
-                    self.infoLabel?.isHidden = true
-                    self.connectToRoom(name: "conference", encodingParameters: encodingParams)
-
-                    let playVideo = UIBarButtonItem(title: "Play Video", style: .plain, target: self, action: #selector(self.pickDocument(_:)))
-                    let browseWeb = UIBarButtonItem(title: "Browse Web", style: .plain, target: self, action: #selector(self.browseWeb(_:)))
-                    self.navigationItem.leftBarButtonItem = playVideo
-                    self.navigationItem.rightBarButtonItem = browseWeb
-                }
+            self.conferenceButton?.isEnabled = true
```

Comment:
And completion here is also on the main thread, correct?

Reply:
Yes so much easier now. 😄